pydantic_factories/factory.py

```diff
@@ -230,6 +230,17 @@ def _get_faker(cls) -> Faker:
             return cls.__faker__
         return default_faker
 
+    @classmethod
+    def _should_use_alias_name(cls, model_field: ModelField, model: Type[T]) -> bool:
+        """Determines whether a given model field should be set by an alias"""
+        if not model_field.alias:
+            return False
+
+        if issubclass(model, BaseModel) and model.__config__.allow_population_by_field_name:
```

Comment:
There is no need to recheck for `BaseModel`, it's already done in `_get_model`.

Reply:
`_get_model` doesn't cast vanilla dataclasses to `BaseModel`. 

As I can see, only if a model is already pydantic model, we do some things. Vanilla dataclasses stay the same
```python
@classmethod
    def _get_model(cls) -> Type[T]:
        """
        Returns the factory's model
        """
        if not hasattr(cls, "__model__") or not cls.__model__:
            raise ConfigurationError("missing model class in factory Meta")
        model = cls.__model__
        if is_pydantic_model(model): <----- datacalsses is not pydantic model
            with suppress(NameError):
                cast(BaseModel, model).update_forward_refs()
        return model
```
And I found it when I test it.

Reply:
Yes, why not use the helper `is_pydantic_model` then?

Reply:
Case mypy can't understand what `is_pydantic_model` have `issubclass` check inside and expect T class in the second condition

```
error: "Type[T]" has no attribute "__config__"
```